app/upgrades/v4/upgrades.go

```diff
@@ -0,0 +1,119 @@
+package v4
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	comettypes "github.com/cometbft/cometbft/proto/tendermint/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	consensuskeeper "github.com/cosmos/cosmos-sdk/x/consensus/keeper"
+	"github.com/cosmos/cosmos-sdk/x/consensus/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/quasar-finance/quasar/app/keepers"
+	slinkyconstants "github.com/skip-mev/slinky/cmd/constants/marketmaps"
+	marketmapkeeper "github.com/skip-mev/slinky/x/marketmap/keeper"
+	marketmaptypes "github.com/skip-mev/slinky/x/marketmap/types"
+)
+
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	keepers *keepers.AppKeepers,
+) upgradetypes.UpgradeHandler {
+	return func(c context.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		ctx := sdk.UnwrapSDKContext(c)
+
+		ctx.Logger().Info("Starting module migrations...")
+		vm, err := mm.RunMigrations(ctx, configurator, fromVM)
+		if err != nil {
+			return vm, err
+		}
+
+		ctx.Logger().Info("Setting consensus params...")
+		err = enableVoteExtensions(ctx, keepers.ConsensusParamsKeeper)
+		if err != nil {
+			return nil, err
+		}
+
+		ctx.Logger().Info("Setting marketmap params...")
+		err = setMarketMapParams(ctx, keepers.MarketmapKeeper)
+		if err != nil {
+			return nil, err
+		}
+
+		ctx.Logger().Info("Setting marketmap and oracle state...")
+		err = setMarketState(ctx, keepers.MarketmapKeeper)
+		if err != nil {
+			return nil, err
+		}
+
+		ctx.Logger().Info(fmt.Sprintf("Migration {%s} applied", UpgradeName))
+		return vm, nil
+	}
+}
+
+func setMarketMapParams(ctx sdk.Context, marketmapKeeper *marketmapkeeper.Keeper) error {
+	marketmapParams := marketmaptypes.Params{
+		MarketAuthorities: []string{authtypes.NewModuleAddress(govtypes.ModuleName).String()}, //MarketMapAuthorityMultisig},
+		Admin:             authtypes.NewModuleAddress(govtypes.ModuleName).String(),           // TODO: decide admin
```

Comment:
todo

Reply:
I would say okay to keep this as gov module.